src/validate.c

```diff
@@ -109,38 +110,57 @@ int main(int argc,char **argv)	{
 
     char *filename = argv[1]; 
 
-    unsigned long skip_offset = 0;
-    unsigned long skip_length = 0;
+    unsigned long skip_offset_sig = 0;
+    unsigned long skip_length_sig = 0;
+
+    unsigned long skip_offset_key = 0;
+    unsigned long skip_length_key = 0;
   
-    if (!appimage_get_elf_section_offset_and_length(filename, ".sha256_sig", &skip_offset, &skip_length)) {
+    if (!appimage_get_elf_section_offset_and_length(filename, ".sha256_sig", &skip_offset_sig, &skip_length_sig)) {
         fprintf(stderr, "Failed to read .sha256_sig section");
         exit(1);
     }
+    if (!appimage_get_elf_section_offset_and_length(filename, ".sig_key", &skip_offset_key, &skip_length_key)) {
```

Comment:
Older AppImages simply don't have this section at all. When AppImageUpdate gained some actual signature verification support, there was a need for the pubkey to be distributed along with the AppImages. Before, that wasn't needed.

I think if we fix this, there should be a fallback for these older runtimes. I could however also live with this tool not supporting old AppImages any more. (The old code can always be restored from the repo, after all.)

@probonopd what do you think?